tests/test_unittest/test_gtc/test_passes/test_oir_optimizations/test_horizontal_execution_merging.py

```diff
@@ -79,6 +81,35 @@ def test_horiz_exec_merging_map_scalar():
     assert len(hexecs[0].declarations) == 2
 
 
+def test_horiz_exec_merging_complexity():
+    n = 1000
+    testee = StencilFactory(
+        vertical_loops__0__sections__0__horizontal_executions=[
+            HorizontalExecutionFactory(
+                body=[AssignStmtFactory(left__name="tmp0", right__name="input")]
+            )
+        ]
+        + [
+            HorizontalExecutionFactory(
+                body=[AssignStmtFactory(left__name=f"tmp{i + 1}", right__name=f"tmp{i}")]
+            )
+            for i in range(n)
+        ]
+        + [
+            HorizontalExecutionFactory(
+                body=[AssignStmtFactory(left__name="output", right__name=f"tmp{n}")]
+            ),
+        ],
+        declarations=[TemporaryFactory(name=f"tmp{i}") for i in range(n)],
+    )
+    start_time = time.process_time()
+    transformed = HorizontalExecutionMerging().visit(testee)
+    process_time = time.process_time() - start_time
+    assert process_time < 5
```

Comment:
`n` is chosen to get a reasonable run time for the test with the fix.On my laptop it is slightly less than one second and more than 40 seconds without the fix. So hopefully the 5 seconds are a reasonable amount of time to check for failure.